cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -337,10 +337,13 @@ struct SparkMurmurHash3_32 {
   template <typename TKey>
   result_type CUDA_DEVICE_CALLABLE compute(TKey const& key) const
   {
-    constexpr int len        = sizeof(TKey);
-    int8_t const* const data = reinterpret_cast<int8_t const*>(&key);
-    constexpr int nblocks    = len / 4;
+    return compute_bytes(reinterpret_cast<int8_t const*>(&key), sizeof(TKey));
+  }
 
+  result_type CUDA_DEVICE_CALLABLE compute_bytes(int8_t const* const data,
+                                                 cudf::size_type const len) const
+  {
+    int32_t nblocks       = len / 4;
```

Comment:
```suggestion
    int32_t const nblocks       = len / 4;
```
And similar for the line below it.

Reply:
`h1` -- line 347 cannot be made const because it's the working hash value. There are a few other lines that